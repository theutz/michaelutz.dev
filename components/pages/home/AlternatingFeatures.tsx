/* eslint-disable @next/next/no-img-element */
import Image from "next/image";
import { ThumbUpIcon, GlobeIcon } from "@heroicons/react/outline";
import {
  useEndorsement,
  Endorsement as EndorsementType,
} from "hooks/useEndorsement";
import { useState } from "react";
import { classNames } from "utils/classNames";

function Endorsement({
  name,
  title,
  company,
  content,
  image,
}: EndorsementType) {
  const [showFull, setShowFull] = useState(false);
  return (
    <blockquote>
      <div onClick={() => setShowFull((prev) => !prev)}>
        <p
          className={classNames(
            "text-base text-gray-500 block whitespace-pre-line cursor-pointer",
            !showFull && "line-clamp-3"
          )}
        >
          &ldquo;{content.trim()}&rdquo;
        </p>
      </div>
      <footer className="mt-3">
        <div className="flex items-center space-x-3">
          <div className="relative flex-shrink-0">
            <div className="w-6 h-6">
              <Image
                className="rounded-full"
                layout="fill"
                objectFit="contain"
                src={image}
                alt={name}
              />
            </div>
          </div>
          <div className="text-base font-medium text-gray-700">
            {name}, {title} @ {company}
          </div>
        </div>
      </footer>
    </blockquote>
  );
}

export function AlternatingFeatureSections() {
  const { endorsement, increment } = useEndorsement();
  return (
    <div className="relative pt-16 pb-32 overflow-hidden">
      <div
        aria-hidden="true"
        className="absolute inset-x-0 top-0 h-48 bg-gradient-to-b from-gray-100"
      />
      <div className="relative">
        <div className="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24">
          <div className="max-w-xl px-4 mx-auto sm:px-6 lg:py-16 lg:max-w-none lg:mx-0 lg:px-0">
            <div>
              <div>
                <span className="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600">
                  <ThumbUpIcon
                    className="w-6 h-6 text-white"
                    aria-hidden="true"
                  />
                </span>
              </div>
              <div className="mt-6">
                <h2 className="text-3xl font-extrabold tracking-tight text-gray-900">
                  The Humans Have Spoken
                </h2>
                <p className="mt-4 text-lg text-gray-500">
                  Building tech products is also about building relationships.
                  Here&apos;s what some of my co-workers have to say about
                  working with me.
                </p>
                <div className="mt-6">
                  <button
                    onClick={() => increment()}
                    className="flex inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                  >
                    Read Another
                  </button>
                </div>
              </div>
            </div>
            <div className="pt-6 mt-8 border-t border-gray-200">
              <Endorsement {...endorsement} />
            </div>
          </div>
          <div className="mt-12 sm:mt-16 lg:mt-0">
            <div className="pl-4 -mr-48 sm:pl-6 md:-mr-16 lg:px-0 lg:m-0 lg:relative lg:h-full">
              <div className="relative lg:m-0 lg:h-full">
                <img
                  className="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:left-0 lg:h-full lg:w-auto lg:max-w-none filter brightness-90"
                  src="/assets/images/seattle-skyline.jpg"
                  alt="Inbox user interface"
                />
                <div className="absolute inset-0 mix-blend-overlay bg-gradient-to-r from-purple-800 to-indigo-600 lg:w-massive"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="mt-24">
        <div className="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24">
          <div className="max-w-xl px-4 mx-auto sm:px-6 lg:py-32 lg:max-w-none lg:mx-0 lg:px-0 lg:col-start-2">
            <div>
              <div>
                <span className="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600">
                  <GlobeIcon
                    className="w-6 h-6 text-white"
                    aria-hidden="true"
                  />
                </span>
              </div>
              <div className="mt-6">
                <h2 className="text-3xl font-extrabold tracking-tight text-gray-900">
                  Better understand your developer
                </h2>
                <p className="mt-4 text-lg text-gray-500">
                  I was born in lovely Seattle, Washington in the USA. I met my
                  wife in Nashville, Tennessee, and we&apos;ve moved to
                  Helsinki, Finland to start a family.
                </p>
                <div className="mt-6">
                  <a
                    href="#"
                    className="inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                  >
                    Get started
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div className="mt-12 sm:mt-16 lg:mt-0 lg:col-start-1">
            <div className="pr-4 -ml-48 sm:pr-6 md:-ml-16 lg:px-0 lg:m-0 lg:relative lg:h-full">
              <div className="relative lg:h-full lg:m-0">
                <img
                  className="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:right-0 lg:h-full lg:w-auto lg:max-w-none"
                  src="/assets/images/helsinki-skyline.jpg"
                  alt="Customer profile user interface"
                />
                <div className="absolute inset-0 lg:left-auto mix-blend-overlay bg-gradient-to-r from-purple-800 to-indigo-600 lg:w-massive"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default AlternatingFeatureSections;
