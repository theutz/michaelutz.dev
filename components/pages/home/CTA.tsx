export function CTA() {
  return (
    <div className="bg-white">
      <div className="max-w-4xl px-4 py-16 mx-auto sm:px-6 sm:py-24 lg:max-w-7xl lg:px-8 lg:flex lg:items-center lg:justify-between">
        <h2 className="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-4xl">
          <span className="block">Ready to get started?</span>
          <span className="block text-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-clip-text">
            Get in touch or create an account.
          </span>
        </h2>
        <div className="mt-6 space-y-4 sm:space-y-0 sm:flex sm:space-x-5">
          <a
            href="#"
            className="flex items-center justify-center px-4 py-3 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
          >
            Learn more
          </a>
          <a
            href="#"
            className="flex items-center justify-center px-4 py-3 text-base font-medium text-indigo-800 border border-transparent rounded-md shadow-sm bg-indigo-50 hover:bg-indigo-100"
          >
            Get started
          </a>
        </div>
      </div>
    </div>
  );
}

export default CTA;
