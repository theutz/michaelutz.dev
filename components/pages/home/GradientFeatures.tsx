import {
  BeakerIcon,
  ChatIcon,
  DatabaseIcon,
  DocumentDuplicateIcon,
  HomeIcon,
  QuestionMarkCircleIcon,
  SparklesIcon,
  TerminalIcon,
} from "@heroicons/react/outline"

const features = [
  {
    name: "The Basics",
    description:
      "HTML, CSS, and JavaScript. The world runs on this stuff, and I've been using it for decades.",
    icon: HomeIcon,
  },
  {
    name: "Front-End Frameworks",
    description:
      "Tons of experience with React, and some experience with Angular and Vue. I'd love to learn Svelte.",
    icon: SparklesIcon,
  },
  {
    name: "Speaking of React...",
    description:
      "Familiar with all the old favorites like Redux & Enzyme, as well as all the newest goodies.",
    icon: ChatIcon,
  },
  {
    name: "What About The Backend?",
    description:
      "I can write a mean backend (pun intended), but I like PostgreSQL and TypeScript for sanity.",
    icon: DatabaseIcon,
  },
  {
    name: "Can you do WordPress?",
    description:
      "Sure! I've been writing WordPress sites since 2008 or so. PHP & MySQL are my friends, too.",
    icon: DocumentDuplicateIcon,
  },
  {
    name: "Can you write tests?",
    description:
      "I love writing tests! Unit testing with Jest and integration/end-to-end testing with Cypress.",
    icon: BeakerIcon,
  },
  {
    name: "So, you just do tech stuff?",
    description:
      "Not at all! I have a passion and eye for design, user experience, and accessibility.",
    icon: TerminalIcon,
  },
  {
    name: "How did you make this site?",
    description:
      "I work smarter not harder. This site is made and deployed with Tailwind CSS & UI, NextJS, and Vercel.",
    icon: QuestionMarkCircleIcon,
  },
]

export function GradientFeatures() {
  return (
    <div className="bg-gradient-to-r from-purple-800 to-indigo-700">
      <div className="max-w-4xl px-4 py-16 mx-auto sm:px-6 sm:pt-20 sm:pb-24 lg:max-w-7xl lg:pt-24 lg:px-8">
        <h2 className="text-3xl font-extrabold tracking-tight text-white">
          Skills you&apos;ll definitely want
        </h2>
        <p className="max-w-3xl mt-4 text-lg text-purple-200">
          I&apos;ve been making web stuff for decades. I&apos;ve got experience
          with multiple content management systems, and I love the new JAMstack
          stuff!
        </p>
        <div className="mt-12 grid grid-cols-1 gap-x-6 gap-y-12 sm:grid-cols-2 lg:mt-16 lg:grid-cols-4 lg:gap-x-8 lg:gap-y-16">
          {features.map((feature) => (
            <div key={feature.name}>
              <div>
                <span className="flex items-center justify-center w-12 h-12 bg-white rounded-md bg-opacity-10">
                  <feature.icon
                    className="w-6 h-6 text-white"
                    aria-hidden="true"
                  />
                </span>
              </div>
              <div className="mt-6">
                <h3 className="text-lg font-medium text-white">
                  {feature.name}
                </h3>
                <p className="mt-2 text-base text-purple-200">
                  {feature.description}
                </p>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}

export default GradientFeatures
