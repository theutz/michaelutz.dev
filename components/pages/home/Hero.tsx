/* eslint-disable @next/next/no-img-element */

export function Hero() {
  return (
    <div className="relative">
      <div className="absolute inset-x-0 bottom-0 bg-gray-100 h-1/2" />
      <div className="mx-auto max-w-7xl sm:px-6 lg:px-8">
        <div className="relative shadow-xl sm:rounded-2xl sm:overflow-hidden">
          <div className="absolute inset-0">
            <img
              className="object-cover w-full h-full"
              src="https://images.unsplash.com/photo-1521737852567-6949f3f9f2b5?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2830&q=80&sat=-100"
              alt="People working on laptops"
            />
            <div className="absolute inset-0 bg-gradient-to-r from-purple-800 to-indigo-700 mix-blend-multiply" />
          </div>
          <div className="relative px-4 py-16 sm:px-6 sm:py-24 lg:py-32 lg:px-8">
            <h1 className="text-4xl font-extrabold tracking-tight text-center sm:text-5xl lg:text-6xl">
              <span className="block text-white">Not Your Average</span>
              <span className="block text-indigo-200">Web Dude</span>
            </h1>
            <p className="max-w-lg mx-auto mt-6 text-xl text-center text-indigo-200 sm:max-w-3xl">
              I&apos;m passionate about creating experiences for the web that
              are useful, enjoyable, and intuitive.
            </p>
            <div className="max-w-sm mx-auto mt-10 sm:max-w-none sm:flex sm:justify-center">
              <div className="space-y-4 sm:space-y-0 sm:mx-auto sm:inline-grid sm:grid-cols-2 sm:gap-5">
                <a
                  href="#"
                  className="flex items-center justify-center px-4 py-3 text-base font-medium text-indigo-700 bg-white border border-transparent rounded-md shadow-sm hover:bg-indigo-50 sm:px-8"
                >
                  Learn More
                </a>
                <a
                  href="#"
                  className="flex items-center justify-center px-4 py-3 text-base font-medium text-white bg-indigo-500 border border-transparent rounded-md shadow-sm bg-opacity-60 hover:bg-opacity-70 sm:px-8"
                >
                  Reach Out
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Hero;
