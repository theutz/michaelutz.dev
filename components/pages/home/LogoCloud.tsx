/* eslint-disable @next/next/no-img-element */
import Image, { ImageProps } from "next/image";
import unityLogo from "public/assets/images/unity-logo.png";
import smartlyLogo from "public/assets/images/smartly-logo.png";
import quenticLogo from "public/assets/images/quentic-logo.png";
import uhsLogo from "public/assets/images/uhs-logo.png";
import ersLogo from "public/assets/images/ers-logo.png";
import { classNames } from "utils/classNames";

export const companies: readonly Pick<ItemProps, "src" | "alt">[] = [
  { src: unityLogo, alt: "Unity" },
  { src: smartlyLogo, alt: "Smartly.io" },
  { src: quenticLogo, alt: "Quentic" },
  { src: uhsLogo, alt: "Universal Hospital Systems" },
  { src: ersLogo, alt: "Emergency Reporting" },
];

export function LogoCloud() {
  return (
    <div className="bg-gray-100" data-cy="logo-cloud">
      <div className="px-4 py-16 mx-auto max-w-7xl sm:px-6 lg:px-8">
        <p className="text-sm font-semibold tracking-wide text-center text-gray-500 uppercase">
          Trusted by a handful of former employers
        </p>
        <div
          data-cy="logo-cloud-container"
          className="mt-6 grid grid-cols-2 gap-8 md:grid-cols-6 lg:grid-cols-5"
        >
          {companies.map((company, i) => (
            <Item
              key={company.alt}
              {...company}
              current={i + 1}
              length={companies.length}
            />
          ))}
        </div>
      </div>
    </div>
  );
}

type ItemProps = {
  src: Exclude<ImageProps["src"], string>;
  alt: ImageProps["alt"];
  current: number;
  length: number;
};

function Item({ src, alt, current, length }: ItemProps) {
  const isLast = current === length;
  const is2ndToLast = length - 1 === current;

  return (
    <div
      className={classNames(
        "relative filter grayscale opacity-40 h-12 md:col-span-2 flex justify-center lg:col-span-1",
        isLast ? "col-span-2 md:col-start-4" : "col-span-1",
        is2ndToLast && "md:col-start-2"
      )}
    >
      <Image layout="fill" objectFit="contain" src={src} alt={alt} />
    </div>
  );
}

export default LogoCloud;
