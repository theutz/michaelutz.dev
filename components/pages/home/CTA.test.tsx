import { render } from "@testing-library/react";
import { CTA } from "components/pages/home/CTA";

describe("<CTA />", () => {
  it("renders", () => {
    // Arrange

    // Act
    const { asFragment } = render(<CTA />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
