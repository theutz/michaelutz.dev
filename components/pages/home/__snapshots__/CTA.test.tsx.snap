// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CTA /> renders 1`] = `
<DocumentFragment>
  <div
    class="bg-white"
  >
    <div
      class="max-w-4xl px-4 py-16 mx-auto sm:px-6 sm:py-24 lg:max-w-7xl lg:px-8 lg:flex lg:items-center lg:justify-between"
    >
      <h2
        class="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-4xl"
      >
        <span
          class="block"
        >
          Ready to get started?
        </span>
        <span
          class="block text-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-clip-text"
        >
          Get in touch!
        </span>
      </h2>
      <div
        class="mt-6 space-y-4 sm:space-y-0 sm:flex sm:space-x-5"
      >
        <a
          class="flex items-center justify-center px-4 py-3 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
          href="#"
        >
          Contact Me
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
